src/imageunpacker/serviceimage.cpp

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <filesystem>
+#include <fstream>
+#include <optional>
+#include <regex>
+#include <stdexcept>
+#include <vector>
+
+#include <Poco/DigestEngine.h>
+#include <Poco/DigestStream.h>
+#include <Poco/SHA2Engine.h>
+#include <Poco/StreamCopier.h>
+
+#include <utils/filesystem.hpp>
+#include <utils/image.hpp>
+#include <utils/json.hpp>
+
+#include "logger/logmodule.hpp"
+#include "serviceimage.hpp"
+
+namespace fs = std::filesystem;
+
+/***********************************************************************************************************************
+ * Consts
+ **********************************************************************************************************************/
+
+namespace {
+const char* cBlobsFolder  = "blobs";
+const char* cManifestFile = "manifest.json";
+const char* cTmpRootFSDir = "tmprootfs";
+const int   cBufferSize   = 1024 * 1024; // 1MB
+} // namespace
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+struct Descriptor {
+    aos::common::utils::Digest mDigest;
+};
+
+struct Manifest {
+    Descriptor              mConfig;
+    std::vector<Descriptor> mLayers;
+};
+
+struct ServiceManifest {
+    std::optional<Descriptor> mAosService;
+    Manifest                  mManifest;
+};
+
+struct ImageParts {
+    std::string mImageConfigPath;
+    std::string mServiceConfigPath;
+    std::string mServiceFSPath;
+};
+
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+static Manifest ParseManifest(const aos::common::utils::CaseInsensitiveObjectWrapper& object)
+{
+    LOG_DBG() << "Parsing manifest";
+
+    Manifest manifest;
+
+    manifest.mConfig.mDigest = object.GetObject("config").GetValue<std::string>("digest");
```

Comment:
Exception is to thrown if "config" is not in JSON
Consider improving error handling to simplify failure investigation

Reply:
This exception will be handled in the UnpackService function, and we can retrieve the message through the exception object.